### DISCUSSION TITLE
Several warning popups

### DISCUSSION THREAD

**Discussion by ElectronMiku** [drizzle-mizzle/CharacterAI-Discord-Bot#48] [Q&A]
When opening the .exe i get 10 exactly the same popups right after closing each of them: 
![image](https://user-images.githubusercontent.com/89693828/229220818-6bee0d0c-a4ad-4432-b4ce-e2aa399811c7.png)
"could not start the program correctly"

my cmd: 
![image](https://user-images.githubusercontent.com/89693828/229220923-eda20aad-c136-442f-b933-fd67a01c7490.png)

sending the `find` command makes the bot start typing but not send anything, and then stops typing. nothing happens in cmd during that time.
also made sure the tokens are correctly pasted.

**Comment by drizzle-mizzle** [comment]
Woah...🫥
Did you try to run it as Administrator?

**Reply by ElectronMiku** [reply]
i did. no difference.

**Reply by drizzle-mizzle** [reply]
Do you have antivirus? Especially, `NOD32`? That's what I found: [link 1](https://support.google.com/chrome/thread/158493819/google-chrome-%D0%BD%D0%B5-%D0%B7%D0%B0%D0%BF%D1%83%D1%81%D0%BA%D0%B0%D0%B5%D1%82%D1%81%D1%8F-%D0%BF%D1%80%D0%B8-%D0%B7%D0%B0%D0%BF%D1%83%D1%81%D0%BA%D0%B5-%D0%BF%D0%B8%D1%88%D0%B5%D1%82-%D0%BE%D1%88%D0%B8%D0%B1%D0%BA%D0%B0-%D0%BF%D1%80%D0%B8-%D0%B7%D0%B0%D0%BF%D1%83%D1%81%D0%BA%D0%B5-%D0%BF%D1%80%D0%B8%D0%BB%D0%BE%D0%B6%D0%B5%D0%BD%D0%B8%D1%8F-0xc0000022?hl=ru), [link 2](https://answers.microsoft.com/ru-ru/microsoftedge/forum/all/%D0%BE%D1%88%D0%B8%D0%B1%D0%BA%D0%B0-0xc0000022/bb626975-049e-4ac4-918b-9ec3b7db6ec3)
People say that NOD32 has some "browser defense" utility that may brake some Chrome installations.

**Reply by ElectronMiku** [reply]
i use the windows defender one. disabled all of its safety thingies but i still got the popups. 

whats also weird tho is that after starting up my pc and opening it the first time, there were no popups (still no response from the bot when sending the `find` command). 
but closing it and opening it again made it show the pop ups. 
it feels inconsistent cuz i had to reopen it more than once to get the errors.

**Reply by drizzle-mizzle** [reply]
Well, seems like it's some problem with your OS in the first place.
Maybe some windows update could corrupt it, maybe it's some application on auto-startup (the fact, that it kinda worked in the moment after reboot, hints to it). 
All I can recommend now is to try `sfc/scannow` and/or check for new Windows updates.
Alternative, you can try to set it up on WSL: #41

**Reply by ElectronMiku** [reply]
i updated my windows (not re-installing).
disabled auto-start on windows defender.
performed `sfc/scannow`

none of these worked but setting it up on WSL did (almost).
it shows that its launching the browser and downloading the chrome executable, but its been stuck on that for more than 15 minutes.
in the windows version it had the green "Running" text but in the WSL thing it never came up.
i tried disabling all autostart apps and then ran it again. same result there and `find` still doesnt work.

**Reply by drizzle-mizzle** [reply]
Try to download .dll from [here](https://github.com/drizzle-mizzle/CharacterAI.Net/releases/tag/release) and place it right in the bot folder in linux version, replacing existing one:

![image](https://user-images.githubusercontent.com/55811932/229292008-2c77c006-88f9-43fd-bf44-8dab7e757a6b.png)

I've updated it a little, so it will now catch any occuring errors while launching puppeteer.

Also, can you check, if your chrome executable has been unzipped? Go to the bot folder -> puppeteer-chrome. What lies there, zip archive or a folder? If it's a folder, can you show me it's full path to executable, it should be similar to this:

![image](https://user-images.githubusercontent.com/55811932/229292602-99e3ad85-c72d-4442-97fb-05d393e9af3f.png)

**Reply by ElectronMiku** [reply]
downloaded the .dll and replaced it. still no difference in the terminal.

this is how my full path looks like.
![image](https://user-images.githubusercontent.com/89693828/229293228-5174196d-1530-4048-88c0-58da3f876b7f.png)

**Reply by drizzle-mizzle** [reply]
Try again, [I made another changes](https://github.com/drizzle-mizzle/CharacterAI.Net/releases/download/release/CharacterAI.dll)

**Reply by drizzle-mizzle** [reply]
And delete whole `puppeteer-chrome` folder before the launch

**Reply by ElectronMiku** [reply]
replaced the dll again and deleted the folder before launch.

downloading the browser went ok but then when it launched:

```
Launching browser... PuppeteerSharp.ProcessException: Failed to launch browser! /mnt/d/downloads/cAI-Discord-Linux-x64/puppeteer-chrome/Linux-1069273/chrome-linux/chrome: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory

   at PuppeteerSharp.States.ChromiumStartingState.StartCoreAsync(LauncherBase p) in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\States\ChromiumStartingState.cs:line 83
   at PuppeteerSharp.States.ChromiumStartingState.StartCoreAsync(LauncherBase p) in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\States\ChromiumStartingState.cs:line 89
   at PuppeteerSharp.Launcher.LaunchAsync(LaunchOptions options) in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\Launcher.cs:line 62
   at PuppeteerSharp.Launcher.LaunchAsync(LaunchOptions options) in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\Launcher.cs:line 93
   at CharacterAI.Integration.LaunchChromeAsync()
```

**Reply by drizzle-mizzle** [reply]
That's it.
Execute `sudo apt update && sudo apt upgrade && sudo apt install libnss3-dev libgdk-pixbuf2.0-dev libgtk-3-dev libxss-dev` and launch it again.

**Reply by ElectronMiku** [reply]
got a similar error but with different libraries, posting the whole error just in case:

```
Launching browser... PuppeteerSharp.ProcessException: Failed to launch browser! /mnt/d/downloads/cAI-Discord-Linux-x64/puppeteer-chrome/Linux-1069273/chrome-linux/chrome: error while loading shared libraries: libasound.so.2: cannot open shared object file: No such file or directory

   at PuppeteerSharp.States.ChromiumStartingState.StartCoreAsync(LauncherBase p) in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\States\ChromiumStartingState.cs:line 83
   at PuppeteerSharp.States.ChromiumStartingState.StartCoreAsync(LauncherBase p) in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\States\ChromiumStartingState.cs:line 89
   at PuppeteerSharp.Launcher.LaunchAsync(LaunchOptions options) in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\Launcher.cs:line 62
   at PuppeteerSharp.Launcher.LaunchAsync(LaunchOptions options) in C:\projects\puppeteer-sharp\lib\PuppeteerSharp\Launcher.cs:line 93
   at CharacterAI.Integration.LaunchChromeAsync()
```

i guess i should do the same thing again but with every library that it says is missing?

**Reply by drizzle-mizzle** [reply]
yep, I've found a full dependencies list in their doc:
`sudo apt update && sudo apt upgrade && sudo apt install libgtk-3-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2`

**Reply by ElectronMiku** [reply]
cool, did that and got no more errors anywhere. still no results from using `find` tho.

**Reply by drizzle-mizzle** [reply]
So now you have that log?

![image](https://user-images.githubusercontent.com/55811932/229299542-8b545649-13bb-4191-a36c-d69869b389d3.png)

Does simple `set` command work?

**Reply by ElectronMiku** [reply]
oh yeah! it works now! it didnt work when i did `find` but `set` did! 🎉

**Reply by drizzle-mizzle** [reply]
Still strange it didn't though... On my side it does. 
Gotta investigate it. If you'll find some pattern in this behavior, please tell me.

**Reply by ElectronMiku** [reply]
sure thing, will keep u updated

**Comment by drizzle-mizzle** [comment]
Also, does it still works when you close all those errors? Try to execute some command, like `find`

**Reply by ElectronMiku** [reply]
oh yeah, forgot to mention about that. using `find` yields nothing. it says its typing for a moment but nothing else happens, it eventually stops typing. and nothing happens in the cmd either. pinging works tho.
